### PR TITLE
stdbuf: remove crash macro

### PR DESCRIPTION
**Pull request by cswn**
This is related to issue: https://github.com/uutils/coreutils/issues/5487
Removes the `crash!` macro from `stdbuf`.

### CODE REVIEW

**Comment by cakebaker** [comment]
Did you see that `test_stdbuf_invalid_mode_fails` now fails?

**Comment by cswn** [comment]
No, I didn't, will take a look now

File: src/uu/stdbuf/src/stdbuf.rs [hunk 1]
```diff
@@ -102,7 +103,10 @@ fn check_option(matches: &ArgMatches, name: &str) -> Result<BufferType, ProgramO
                 }
             }
             x => parse_size_u64(x).map_or_else(
-                |e| crash!(125, "invalid mode {}", e),
+                |e| {
+                    set_exit_code(125);
```
Comment (tertsdiepraam) [inline src/uu/stdbuf/src/stdbuf.rs:107]: I think this lines should not be necessary. The exit code should be added to the error here:

https://github.com/uutils/coreutils/blob/26df9445fb70152abfd523c0bdb75def6c7d5475/src/uu/stdbuf/src/stdbuf.rs#L151

It'd be great if you could also verify that this is actually the case by writing a test for it.

**Review by tertsdiepraam — COMMENTED** [review]
Nice! Just one suggestion

File: src/uu/stdbuf/src/stdbuf.rs (hunk 1 above)
Comment (cswn) [reply src/uu/stdbuf/src/stdbuf.rs:107]: Ah, you're right, I've taken that line out. However, the test already asserts that the error code is 125: https://github.com/cswn/coreutils/blob/26df9445fb70152abfd523c0bdb75def6c7d5475/tests/by-util/test_stdbuf.rs#L67C9-L67C9

**Comment by cakebaker** [comment]
Thanks!

**Pull request merged**